src/test/scala/com/ambiata/poacher/hdfs/Arbitraries.scala

```diff
@@ -5,12 +5,15 @@ import Arbitrary._
 import scalaz._, Scalaz._, effect.IO
 import com.ambiata.mundane.control._
 import com.ambiata.mundane.io._
+import com.ambiata.disorder.GenPlus
 
 object Arbitraries {
   implicit def HdfsTemporaryArbitrary: Arbitrary[HdfsTemporary] = Arbitrary(for {
-    i <- Gen.choose(1, 5)
-    a <- Gen.listOfN(i, Gen.identifier)
-    z = a.mkString("/")
+    z <- arbitrary[SubPath]
     f <- Gen.oneOf("", "/")
-  } yield HdfsTemporary(s"temporary-${java.util.UUID.randomUUID().toString}/" + z + f))
+  } yield HdfsTemporary(s"temporary-${java.util.UUID.randomUUID().toString}/" + z.path + f))
+
+  case class SubPath(path: String)
```

Comment:
Ok, run with what you have will leave to @nhibberd to sort, we need a path arbitrary that is going to work. It troubles me that we don't even know what limitation we are violating with these (same sort of thing happens in saws).